File: DSAlab/19114055-01/P1.cpp
/*
Masthead - 
Owner - Master Mayank Meena 
Enroll no. 19114055, O-3
Code Editor - VSCode 2020
Terminal - WSL Ubuntu 2020
Shell - git bash

Contact - 
Website - https://mayankmeena.netlify.app/
Email - mayank_m@cs.iitr.ac.in

Social -
GitHub - https://github.com/maayami
YouTube - https://www.youtube.com/channel/UCwcdyxP6uk5zso-L4lY4Y8g?view_as=subscriber
Twitter - https://twitter.com/Meina_Mk
*/

#include<bits/stdc++.h>
using namespace std;

void printArray(int arr[], int n)
{
    for(int i=0; i<n; i++)
    {
        cout<<arr[i]<<" ";
    }
    cout<<"\n";
}

void swap(int* a, int* b)
{
    int temp = *a;
    *a = *b;
    *b = temp;
}

int partition(int arr[], int start, int end)
{
    int pivotEle = arr[end]; 
    int i=start-1;
    for(int j=start; j<=end; j++)
    {
        if(arr[j] <= pivotEle){
            i++;
            swap(&arr[i], &arr[j]);
        }
    }
    return i;
}

void quickSort(int arr[], int start, int end)
{
    if(start < end){   
        int pivotIndex = partition(arr, start, end);
        quickSort(arr, start, pivotIndex-1);
        quickSort(arr, pivotIndex+1, end);
    }
}

void solve()
{
    int n;
    cin>>n;
    int arr[n];
    for(int i=0; i<n; i++){
        cin>>arr[i];
    }
    quickSort(arr, 0, n-1);
    printArray(arr, n);
}

int main(int argc, char const *argv[])
{
ios_base::sync_with_stdio(false);
cin.tie(NULL);    
    solve();
    return 0;
}

File: DSAlab/19114055-11/P1.java
/*
Masthead - 
Owner - Mayank Meena 
Enroll no. 19114055
Code Editor - VSCode 2020(1.40.1)
Terminal - Ubuntu Terminal 20.04
Shell - git bash

Contact - 
Website - https://mayankmeena.netlify.app/
Email - mayank_m@cs.iitr.ac.in

Social -
GitHub - https://github.com/maayami
YouTube - https://www.youtube.com/channel/UCwcdyxP6uk5zso-L4lY4Y8g?view_as=subscriber
Twitter - https://twitter.com/Meina_Mk
*/
import java.util.Scanner;

class P1 {
    public static void main(String[] args) {
        Scanner inputsP1 = new Scanner(System.in);
        int vertices = inputsP1.nextInt();
        int edges = inputsP1.nextInt();
        int tempEdgeCount = 0;
        String answer;
        boolean isTwoEdgeGraph = edges<vertices?false:true;

        for (int i = 0; i < vertices; i++) {
            tempEdgeCount = 0;
            if (!isTwoEdgeGraph)
                break;
            for (int j = 0; j < vertices; j++) {
                if (inputsP1.nextInt() == 1 && i !=j)
                    tempEdgeCount++;
            }
            isTwoEdgeGraph = tempEdgeCount > 1 ? true : false;
        }
        answer = isTwoEdgeGraph ? "Yes" : "No";
        System.out.println(answer);
        inputsP1.close();
    }
}
